app.html

```diff
@@ -556,6 +583,40 @@
   return "<div class=edit><span class=title>name<br><input type=text class=myinput id=name_input value='"+value+"' onchange='SaveName()'></span></div>\n";
 }
 
+
+function SaveVariation() {
+  var variation_slider = document.getElementById('variation_slider');
+  console.log("SAVE VARIATION " + variation_slider.value);
+  if (presets[current_preset_num].VARIATION != variation_slider.value) {
+    presets[current_preset_num].VARIATION = variation_slider.value;
+    Send("variation " + variation_slider.value);
+  }
+  var variation_field = document.getElementById('variation_field');
+  if ( variation_field.value != variation_slider.value){
+     variation_field.value = variation_slider.value; //update numeric value when altered by slider
+  }
+  const preset_tags = document.getElementsByClassName('preset');
+  //	SetPreset(current_preset_num) //no need, variation is always updates directly
+  UpdatePresets();  
+}
+
+
+function SetVariation(){
+  var set_variation_field = document.getElementById('variation_field');
+  console.log("SET VARIATION " + set_variation_field.value);
+  var set_variation_slider = document.getElementById('variation_slider');
+  set_variation_slider.value = set_variation_field.value;
+  SaveVariation(); // now do the actual update
+}
+
+function make_variation_field(value) {
+  var ret = "<div class=edit>";
+  ret += "<span class=title>variation<br><br><input type=range min=0 max=32767 class=slider id=variation_slider value='"+value+"' onchange='SaveVariation()'>";
```

Comment:
it does the same as typing in var ### in the serial monitor, so it should control whatever is connected to the variation.
Agreed that it would be very nice to see the actual color.
I mostly use it to reset a variation from the color wheel to exaclty 0 when I want to do a 'factory-reset' of that style.

Reply:
Have you tried it with a style that uses ColorChange<> ?

Reply:
Yes I have,
In stead of using the slider, you can use the input field below the slider and type in the variation that you are after (0,1,2,...)
Its is the same for really fine-tuning your in colorwheel mode, in stead of using the slider, you can input any number to give you more precision in tuning your color.

Reply:
I have added a - and + button before and after the numeric input field to make it more user friendly in case of ColorChange<> styles. Now you can increase and decrease by 1 step just by clicking these buttons.